pkg/mock/definition.go

```diff
@@ -34,6 +34,13 @@ type Response struct {
 	Body string `json:"body"`
 }
 
+type Callback struct {
+	Delay       Delay  `json:"delay"`
+	Url         string `json:"url"`
+	ContentType string `json:"contentType"`
+	Body        string `json:"body"`
+}
+
```

Comment:
I think this struct will be more flexible like this:

```suggestion
type Callback struct {
	Url         string `json:"url"`
	Method    string `json:"method"`  
	Body        string `json:"body"`
	Delay       Delay  `json:"delay"`
	HttpHeaders
}

```

What do you think?

Reply:
Yep agreed - I opened an MVP PR early because I wasn't sure whether there was value in merging an MVP first and then adding the flexibility later. This is definitely the ideal design though. I'm happy to make the changes in this PR.

Reply:
Updated - now uses this struct

Reply:
Cool!! I got it, but if you don't mind I prefer to do some iteration over the PR instead of merging the MVP